sdk/identity/identity/test/public/node/azurePipelinesCredential.spec.ts

```diff
@@ -41,14 +43,42 @@ describe("AzurePipelinesCredential", function () {
       tenantId,
       clientId,
       "existingServiceConnectionId",
-      systemAccessToken,
+      systemAccessToken
     );
     const regExp: RegExp =
-      /AzurePipelinesCredential: Authenticated Failed. Received null token from OIDC request. Response status- 404./;
+      /AuthenticationRequiredError: invalid_client: Error\(s\): 700213 .* AADSTS700213: No matching federated identity record found for presented assertion subject .* Please note that the matching is done using a case-sensitive comparison. Check your federated identity credential Subject, Audience and Issuer against the presented assertion/;
     await assert.isRejected(
       credential.getToken(scope),
       regExp,
-      "error thrown doesn't match or promise not rejected",
+      "error thrown doesn't match or promise not rejected"
+    );
+  });
+
+  it.only("failure includes the expected response headers", async function () {
+    if (!isLiveMode()) {
+      this.skip();
+    }
+    // clientId for above service connection
+    const clientId = process.env.AZURE_SERVICE_CONNECTION_CLIENT_ID!;
+    const systemAccessToken = process.env.SYSTEM_ACCESSTOKEN!;
+    const credential = new AzurePipelinesCredential(
+      tenantId,
+      clientId,
+      "existingServiceConnectionId",
+      systemAccessToken
+    );
+    const regExpHeader1: RegExp =
+      /"x-vss-e2eid":\"[a-fA-F0-9]{8}-([a-f0-9A-F]{4}-){3}[0-9a-fA-F]{12}\"/;
+    const regExpHeader2: RegExp = /"x-msedge-ref\":"[a-zA-Z\s:0-9-]*"/;
+    await assert.isRejected(
+      credential.getToken(scope),
+      regExpHeader1,
+      'error contains the expected header "x-vss-e2eid"'
```

Comment:
This is the message that gets printed out when the assertion _fails_ so it should say something like "expected error to contain the header 'x-vss-e2eid' but it was not found" or something along those lines?

Reply:
oh yes!